src/plots/plots.js

```diff
@@ -1564,15 +1570,20 @@ plots.extendObjectWithContainers = function(dest, src, containerPaths) {
             if(!srcContainer) continue;
 
             destProp = Lib.nestedProperty(dest, containerPaths[i]);
-
             destContainer = destProp.get();
+
             if(!Array.isArray(destContainer)) {
                 destContainer = [];
                 destProp.set(destContainer);
             }
 
             for(j = 0; j < srcContainer.length; j++) {
-                destContainer[j] = plots.extendObjectWithContainers(destContainer[j], srcContainer[j]);
+                var srcObj = srcContainer[j];
+
+                if(srcObj === null) destContainer[j] = {};
```

Comment:
More generally, what should non-plain-object items in layout array containers be coerced to?

See current behavior here: http://codepen.io/etpinard/pen/xRGgwW?

I'd vote for completely skipping over non-plain-object items (i.e. non-plain-object items won't show up in `fullLayout`). Any objections? Some may interpret this change as backward incompatible. @alexcjohnson thoughts?

Currently, updatemenus and sliders skip over non-plain-object `buttons` and `step` items (see [here](https://github.com/plotly/plotly.js/blob/38aa4b3841752adaae010f28ba1647c7a7128107/src/components/updatemenus/defaults.js#L89) and [here](https://github.com/plotly/plotly.js/blob/38aa4b3841752adaae010f28ba1647c7a7128107/src/components/sliders/defaults.js#L108)).